src/rpcwallet.cpp

```diff
@@ -3637,3 +3637,20 @@ UniValue searchdzpiv(const UniValue& params, bool fHelp)
     //todo: better response
     return "done";
 }
+
+UniValue createautomintaddress(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+                "createautomintaddress\n"
+                        "Overview: generates new auto mint address\n"
+                        "\nResult\n"
+                        "\"address\"             160hash of public key\n" +
+                HelpExampleCli("createautomintaddress", "") +
+                HelpExampleRpc("createautomintaddress", ""));
+    LOCK(pwalletMain->cs_wallet);
```

Comment:
Per Warrows' concern, could add `EnsureWalletIsUnlocked();` above the LOCK here to ensure that the wallet is unlocked prior to any code execution.